job-descriptions/software-engineer-security.md

```diff
@@ -0,0 +1,51 @@
+![logo](https://sourcegraph.com/.assets/img/sourcegraph-light-head-logo.svg)
+
+# Software Engineer - Security
```

Comment:
Any thoughts on the Security Engineer title? Are we looking for a security specialist which can also code or a software engineer that can also do security stuff?

I'd say the former, and specifically, a security engineer generalist.

Reply:
> Any thoughts on the Security Engineer title?

I hadn't considered it. I am not sure what the practical difference is, so this is probably more about branding (what will the candidates that we are looking for expect to see) more than anything else.

@christinaforney since you have experience working with people in this role, what do you think?

> Are we looking for a security specialist which can also code or a software engineer that can also do security stuff?

We are looking for what the job description describes :) What is your impression on what we are looking for after reading the job description?

It is important that his person has both: ability/experience making code changes and knowledge/experience with security issues (both of these are mentioned in the qualifications).

Reply:
> We are looking for what the job description describes :) What is your impression on what we are looking for after reading the job description?

Well, sure, my question was meant to discuss that :)

> I think it is important that his person has both: ability/experience making code changes and knowledge/experience with security issues (both of these are mentioned in the qualifications).

Agreed that this person needs to be able to code, but would you say necessarily to the same level as our backend software engineers? Someone having those skills AND being a deep security expert might be a tall order?

Reply:
I've seen both job titles, but Security Engineer is probably more common.

Reply:
> - Skilled at understanding, debugging, and writing Go and TypeScript to identify and fix security issues.
> Someone having those skills AND being a deep security expert might be a tall order?

There is a spectrum here and I am looking for a balance of both. We need someone who has enough experience in the security domain that they can be self directed, and enough development experience that they can do the necessary work themselves. It remains to be seen how easy it will be to hire a person like this, but we shouldn't prematurely compromise what we are looking for.

I am going to stick with this title for now.